.github/workflows/sbt.yml

```diff
@@ -222,3 +222,30 @@ jobs:
             name: flink-${{ matrix.flink }}-unit-test-log
             path: |
                 **/target/test-reports/**
+
+  mr:
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      matrix:
+        java:
+          - 8
+          - 11
```

Comment:
jdk 17?

Reply:
Add jdk 17 is reasonable.

Reply:
<details>

<summary>guice 4.0 does not support JDK17 - InaccessibleObjectException</summary>

### guice 4.0  + JDK17

```java
2023-11-22T10:53:55,768 ERROR [Listener at 0.0.0.0/56065] org.apache.celeborn.mapreduce.v2.app.MRAppMasterWithCeleborn: Error starting MRAppMaster
java.lang.ExceptionInInitializerError: null
	at com.google.inject.internal.cglib.reflect.$FastClassEmitter.<init>(FastClassEmitter.java:67) ~[guice-4.0.jar:?]
	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.generateClass(FastClass.java:72) ~[guice-4.0.jar:?]
	at com.google.inject.internal.cglib.core.$DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25) ~[guice-4.0.jar:?]
	at com.google.inject.internal.cglib.core.$AbstractClassGenerator.create(AbstractClassGenerator.java:216) ~[guice-4.0.jar:?]
	at com.google.inject.internal.cglib.reflect.$FastClass$Generator.create(FastClass.java:64) ~[guice-4.0.jar:?]
	at com.google.inject.internal.BytecodeGen.newFastClass(BytecodeGen.java:204) ~[guice-4.0.jar:?]
	at com.google.inject.internal.ProviderMethod$FastClassProviderMethod.<init>(ProviderMethod.java:256) ~[guice-4.0.jar:?]
	at com.google.inject.internal.ProviderMethod.create(ProviderMethod.java:71) ~[guice-4.0.jar:?]
	at com.google.inject.internal.ProviderMethodsModule.createProviderMethod(ProviderMethodsModule.java:275) ~[guice-4.0.jar:?]
	at com.google.inject.internal.ProviderMethodsModule.getProviderMethods(ProviderMethodsModule.java:144) ~[guice-4.0.jar:?]
	at com.google.inject.internal.ProviderMethodsModule.configure(ProviderMethodsModule.java:123) ~[guice-4.0.jar:?]
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340) ~[guice-4.0.jar:?]
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:349) ~[guice-4.0.jar:?]
	at com.google.inject.AbstractModule.install(AbstractModule.java:122) ~[guice-4.0.jar:?]
	at com.google.inject.servlet.ServletModule.configure(ServletModule.java:52) ~[guice-servlet-4.0.jar:?]
	at com.google.inject.AbstractModule.configure(AbstractModule.java:62) ~[guice-4.0.jar:?]
	at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:340) ~[guice-4.0.jar:?]
	at com.google.inject.spi.Elements.getElements(Elements.java:110) ~[guice-4.0.jar:?]
	at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:138) ~[guice-4.0.jar:?]
	at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:104) ~[guice-4.0.jar:?]
	at com.google.inject.Guice.createInjector(Guice.java:96) ~[guice-4.0.jar:?]
	at com.google.inject.Guice.createInjector(Guice.java:73) ~[guice-4.0.jar:?]
	at com.google.inject.Guice.createInjector(Guice.java:62) ~[guice-4.0.jar:?]
	at org.apache.hadoop.yarn.webapp.WebApps$Builder.build(WebApps.java:394) ~[hadoop-yarn-common-3.2.4.jar:?]

Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make protected final java.lang.Class java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain) throws java.lang.ClassFormatError accessible: module java.base does not "opens java.lang" to unnamed module @689349f1
	at java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354) ~[?:?]	
```

</details>

https://issues.apache.org/jira/browse/HADOOP-17177

https://github.com/google/guice/issues/1536

https://github.com/apache/incubator-heron/issues/3790